### PR TITLE
Add SMAPE(symmetric mean absolute percentage error) loss function. 

### PR DESCRIPTION
**Pull request by JinScientist**
Add SMAPE(symmetric mean absolute percentage error) loss function. The existing MAPE has the disadvantage that they put a heavier penalty on positive errors than on negative errors. SMAPE can mitigate this the uneven penalty.

### CODE REVIEW

**Comment by jhseu** [comment]
This also shouldn't be on the 1.1 branch, but I'll let Francois comment first.

**Comment by fchollet** [comment]
We won't merge:

- Not common enough to become part of the core codebase
- Would introduce an API discrepancy with the reference Keras API
